### PR TITLE
rubocop: Formulae with the "lxml" resource have required dependencies

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

- We're moving from `depends_on "python-lxml"` to `resource "lxml" ...` as part of the new Python vendoring plan.
- For `resource "lxml"` to work, `uses_from_macos "libxml2"` and `uses_from_macos "libxslt"` are needed in the formulae.
- This new RuboCop rule enforces that a formula including the "lxml" resource also include those dependencies.

-----

TODO:

- [ ] An autocorrect, probably useful when we start switching these over (right now everything still uses `depends_on "python-lxml"`). Separate PR?

### CODE REVIEW

**Comment by p-linnane** [comment]
Thank you so much for this! We have already moved some formulae over to using the resource, and will be moving the rest over as time allows.

File: Library/Homebrew/rubocops/resource_requires_dependencies.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          resource_nodes = find_every_method_call_by_name(body_node, :resource)
+          lxml = resource_nodes.find { |node| node.arguments.first.str_content == "lxml" }
+          return unless lxml
+
+          uses_from_macos_nodes = find_every_method_call_by_name(body_node, :uses_from_macos)
+          dependencies = uses_from_macos_nodes.map { |node| node.arguments.first.str_content }
+          return if dependencies.include?("libxml2") && dependencies.include?("libxslt")
+
+          offending_node(lxml)
+          problem "Add `uses_from_macos` lines above for \"libxml2\"` and \"libxslt\"."
```
Comment (ZhongRuoyu) [inline Library/Homebrew/rubocops/resource_requires_dependencies.rb:27]:
```suggestion
          problem "Add `uses_from_macos` lines above for `\"libxml2\"` and `\"libxslt\"`."
```

File: Library/Homebrew/test/rubocops/resource_requires_dependencies_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      expect_offense(<<~RUBY)
+        class Foo < Formula
+          url "https://brew.sh/foo-1.0.tgz"
+          homepage "https://brew.sh"
+
+          uses_from_macos "libsomethingelse"
+          uses_from_macos "not_libxml2"
+
+          resource "lxml" do
+          ^^^^^^^^^^^^^^^ FormulaAudit/ResourceRequiresDependencies: Add `uses_from_macos` lines above for "libxml2"` and "libxslt".
```
Comment (ZhongRuoyu) [inline Library/Homebrew/test/rubocops/resource_requires_dependencies_spec.rb:54]:
```suggestion
          ^^^^^^^^^^^^^^^ FormulaAudit/ResourceRequiresDependencies: Add `uses_from_macos` lines above for `"libxml2"` and `"libxslt"`.
```

File: Library/Homebrew/test/rubocops/resource_requires_dependencies_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "rubocops/resource_requires_dependencies"
+
+describe RuboCop::Cop::FormulaAudit::ResourceRequiresDependencies do
```
Comment (ZhongRuoyu) [inline Library/Homebrew/test/rubocops/resource_requires_dependencies_spec.rb:5]:
```suggestion
RSpec.describe RuboCop::Cop::FormulaAudit::ResourceRequiresDependencies do
```

**Review by ZhongRuoyu — COMMENTED** [review]
This is nice. Thanks @issyl0!

**Review by MikeMcQuaid — APPROVED** [review]
Great work as usual @issyl0!

**Pull request merged**